### PR TITLE
[#1205] Domainmodel example: Add duplicated operation validation rule.

### LINKED ISSUE #1205: Domainmodel example: Add duplicated operation validation rule.
See [comment](https://github.com/eclipse/xtext-eclipse/pull/1203#issuecomment-539047064) from @LorenzoBettini

### PR DESCRIPTION
**Pull request by LorenzoBettini**
I started to work on this, starting from the current PR https://github.com/eclipse/xtext-eclipse/pull/1203

I'm planning to add some further tests, but some initial feedback would be helpful :)

Closes #1205

### CODE REVIEW

**Comment by miklossy** [comment]
Thanks for the PR! Could you please rebase it on master?

**Comment by LorenzoBettini** [comment]
I actually merged with master to avoid a push force and leaving a dangling branch in existing local git repositories, is that OK?

**Comment by cdietrich** [comment]
the usual way is to force push on the branch

**Comment by LorenzoBettini** [comment]
So do you want me to push force? Or is the current situation acceptable?

**Comment by cdietrich** [comment]
Yes I would prefer that

**Comment by LorenzoBettini** [comment]
OK, done a push force after rebasing

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend [hunk 1]
```diff
@@ -184,6 +186,54 @@ class ValidationTests {
...
+	}
+
+	@Test def void testDuplicatedOperationWithDifferentSignatureIsAllowed() {
+		val model = '''
+			entity E {
+				op m(int i) {}
+				op m() {}
+			}
+		'''
+		model.parse.assertNoErrors
```
Comment (miklossy) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend]: I prefer not to use the extra local variable `model` if it is used only in once place, but rather inline it:
```

'''
	entity E {
		op m(int i) {}
		op m() {}
	}
'''.parse.assertNoErrors
```

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend [hunk 2]
```diff
@@ -184,6 +186,54 @@ class ValidationTests {
...
+	}
+
+	@Test def void testFeatureAndOperationWithSameNameIsAllowed() {
+		val model = '''
+			entity E {
+				m : String
+				op m() {}
+			}
+		'''
+		model.parse.assertNoErrors
```
Comment (miklossy) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend]: Same here.

**Comment by miklossy** [comment]
@LorenzoBettini The implementation looks fine to me. However, I requested a change to make the test cases more compact and readable. More test cases are welcome, especially with operations having a name more than one letter (to verify that the entire operation names are marked with the error and not only the first letter).

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.xtend [hunk 3]
```diff
@@ -22,6 +22,12 @@ import static org.eclipse.xtext.example.domainmodel.validation.IssueCodes.*
 
 import static extension java.lang.Character.isLowerCase
 import static extension java.lang.Character.isUpperCase
+import com.google.inject.Inject
+import org.eclipse.xtext.xbase.jvmmodel.IJvmModelAssociations
+import org.eclipse.xtext.common.types.JvmGenericType
+import org.eclipse.xtext.xbase.typesystem.^override.OverrideHelper
+import org.eclipse.xtext.common.types.JvmOperation
+import org.eclipse.xtext.example.domainmodel.domainmodel.Operation
```
Comment (miklossy) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.xtend]: Organize imports.

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend (hunk 1 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend]: OK, will do!

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend (hunk 2 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/ValidationTests.xtend]: OK, will do!

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.xtend (hunk 3 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.xtend]: OK, will do

**Comment by LorenzoBettini** [comment]
@miklossy I've update the PR according to your comments, and I also modified a testcase with operations having a name more than one letter (to verify that the entire operation names are marked with the error and not only the first letter)

**Pull request merged**